src/engine/ui/widgets/textField.cxx

```diff
@@ -49,7 +54,7 @@ void TextField::addText(const std::string &text)
 
 std::string TextField::getTextFromID(int id)
 {
-  if (id < _textVector.size())
+  if (static_cast<int>(id) < static_cast<int>(_textVector.size()))
```

Comment:
I think that the call of `static_cast<int>(_textVector.size());` is redundant.
You should declare `int numTextVector = static_cast<int>(_textVector.size());` and use it.

Reply:
i've added a private count variable to the textfield class instead because it's used on multiple locations and added a const function to retrieve it. This could be useful in the future.